xikolo-ios/AppDelegate.swift

```diff
@@ -43,27 +44,65 @@ class AppDelegate : AbstractAppDelegate {
     func application(_ application: UIApplication,
                      continue userActivity: NSUserActivity,
                      restorationHandler: @escaping ([Any]?) -> Void) -> Bool {
+        let url:URL
+        if userActivity.activityType == CSSearchableItemActionType {
+            // This activity represents an item indexed using Core Spotlight, so restore the context related to the unique identifier.
+            // Note that the unique identifier of the Core Spotlight item is set in the activity’s userInfo property for the key CSSearchableItemActivityIdentifier.
+            let uniqueIdentifier = userActivity.userInfo? [CSSearchableItemActivityIdentifier]
+            url = URL.init(string: uniqueIdentifier as! String)!
+            // this contains the courses uuid
+            // Next, find and open the item specified by uniqueIdentifer.
+
+        }else{
+             url = userActivity.webpageURL!
+        }
+        
 
-        guard userActivity.activityType == NSUserActivityTypeBrowsingWeb, let url = userActivity.webpageURL else { return false }
         guard let rootViewController = self.window?.rootViewController as? UITabBarController else {
             print("UITabBarController could not be found")
             return false
         }
-        switch url.lastPathComponent {
-        case "courses":
-            rootViewController.selectedIndex = 1
-        case "dashboard":
-            rootViewController.selectedIndex = 0
-        case "news":
-            rootViewController.selectedIndex = 2
-        default:
-            break
+        //Todo:
+
+
+        //if url.lastPathComponent == "courses" and {
+        //}
+        switch url.pathComponents[1] {
+            case "courses":
+                if url.pathComponents.count > 2{
+                    //support /courses/slug -> course detail page or learning
+                    let slug = url.pathComponents[2]
+                    //get course by slug
+                    //todo the course might not be synced yet, than we could try to fetch from the API by slug
+                    var course: Course?
+                    do {
+                        course = try CourseHelper.getBySlug(slug)
+                    } catch {
+                        course = nil
+                    }
+                    if (course != nil){
+                        self.goToCourse(course!)
+                        return true
+                    }
+
+                }else{
+                    rootViewController.selectedIndex = 1
+                }
+
+            case "dashboard":
+                rootViewController.selectedIndex = 0
+            case "news":
+                rootViewController.selectedIndex = 2
+            default:
+                break
         }
-        
+
         // we can't handle the url, open it with a browser
         let webpageUrl = url
         application.openURL(webpageUrl)
         return false
+
+
```

Comment:
I think @mathebox knows some app for travis. But sometimes added whitespace is intended...